cleanup: fixing (some) special chars

smlng opened the pull request:
Specifically the following things were addressed:
    - replacing special dashes with simple '-'
    - fixing non-ascii whitespaces to plain ' '
    - removing other not correctly displayed chars

    Author names, greek symbols in units, and other commonly used
    UTF8 chars were not adapted.

miri64 commented:
Weird white-spaces I get (and to an extend the dash thing as well), but why do we need to remove the rest? The code is encoded in unicode, not ASCII.

miri64 commented on pkg/openthread/include/ot.h:
```diff
@@ -17,7 +17,7 @@
  * @file
  *
- * @author  Jos� Ignacio Alamos <jialamos@uc.cl>
+ * @author  José Ignacio Alamos <jialamos@uc.cl>
```
?

smlng replied:
the `é` of José looks weird when viewed in Atom or VIM editors, obviously it doesn't on github.

miri64 commented on pkg/emb6/patches/0002-Rename-colliding-files-and-functions.patch:
```diff
@@ -66,7 +66,7 @@ index 2912049..682b7c2 100644
 -#include "random.h"
 +#include "emb6_random.h"
- 
+
```
Changing patch files this way can potentially end badly ;-)

smlng replied:
oops, you're right, that was unintentional. There are other patches with such, which I left out.

kaspar030 commented:
> Author names, greek symbols in units, and other commonly used
> UTF8 chars were not adapted.

I don't get it. What gets replaced with what? I suppose you're replacing wierd ascii with the correct UTF-8?

smlng commented:
I refactored the PR into smaller commits to clarify what is done where, also removed the fixes in that patch file. Btw. this how a part of the diff looks on my terminal (which looks different when viewed on GitHub):

```diff
     mrf24j40_reg_write_long(dev, MRF24J40_REG_RFCON0, channel_value);
     /*
-     * Note:Perform an RF State Machine Reset (see Section 3.1 <93>Reset<94>)
+     * Note: Perform an RF State Machine Reset (see Section 3.1 Reset)
      * after a channel frequency change. Then, delay at least 192 us after
      * the RF State Machine Reset, to allow the RF circuitry to calibrate.
      */
@@ -373,7 +373,7 @@ void mrf24j40_set_option(mrf24j40_t *dev, uint16_t option, bool state)
                 tmp |= MRF24J40_TXMCR_NOCSMA;
                 /* MACMINBE<1:0>: The minimum value of the backoff exponent
                  * in the CSMA-CA algorithm. Note that if this value is set
-                 * to <91>0<92>, collision avoidance is disabled. */
+                 * to 0, collision avoidance is disabled. */
                 mrf24j40_reg_write_short(dev, MRF24J40_REG_TXMCR, tmp);
```

kaspar030 commented:
Did you do this via script or manually?
